src/Meltor.php

```diff
@@ -0,0 +1,872 @@
+     * @param string $file
+     * @return string
+     */
+    public function combinePath(string $folder, string $file): string
+    {
+        return sprintf('%s/%s', $folder, $file);
+    }
+
+    /**
+     * Remove redundant CHARACTER SET definitions from a given structure dump to allow comparison.
+     * Character sets may be defined for a column, but not being extracted when they're identical to the table definition.
+     *
+     * @param string $sqlStructure
+     *
+     * @return string
+     */
+    protected function removeRedundantCharSetFromDump(string $sqlStructure): string
+    {
+        return preg_replace('/CHARACTER SET (\w+)[\s,](?=[^;]+DEFAULT CHARSET=\1\s[^;]+;)/', '', $sqlStructure);
+    }
+
+    /**
+     * Remove generation timestamp from the given structure dump to allow comparison.
+     *
+     * @param string $sqlStructure
+     *
+     * @return string
+     */
+    protected function removeTimeStampFromDump(string $sqlStructure): string
+    {
+        // The time format in mysqldump file footers doesn't pad its zeroes according to ISO 8601, so trying to guess the format.
+        return preg_replace(
+            '/Dump completed on [\d\-\s:]+/',
+            'Dump completed on (removed for comparison)',
+            $sqlStructure
+        );
+    }
+
+    /**
+     * Reads a given database structure file for comparison, and removes incomparable noise from the content.
+     *
+     * The resulting content is only suitable for comparison.
+     *
+     * @param string $structureFileName
+     *
+     * @return string
+     */
+    public function readAndCleanStructure(string $structureFileName): string
+    {
+        $structure = file_get_contents($structureFileName);
+        $structure = $this->removeTimeStampFromDump($structure);
+        $structure = $this->removeRedundantCharSetFromDump($structure);
+
+        return $structure;
+    }
+
+    /**
+     * Writes a new migration into the "migrations" folder.
+     *
+     * Older migrations with the same name will be removed
+     * - in order to keep a current file name to run last
+     * - while not filling the "migrations" folder with each run
+     *
+     * Will return the name of the new migration file.
+     *
+     * @param string $migrationContent
+     * @param string $migrationName
+     * @param int $offset Seconds to add to the file name date
+     * @param Command|null $command Used for shell output
+     *
+     * @return string
+     */
+    public function writeMigration(string $migrationContent, string $migrationName, Command $command = null, int $offset = 0): string
+    {
+        $filename          = sprintf('%s_%s.php', now()->addSeconds($offset)->format('Y_m_d_His'), $migrationName);
+        $migrationFilePath = $this->combinePath($this->config('migration.folder'), $filename);
+
+        $this->removeMigrations($migrationName, $command);
+        $command?->info(sprintf('Generate new migration "%s"', $migrationFilePath));
```

Comment:
Why can't the caller do the output? Then you would also not have to pass in the Command.